core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -156,6 +156,10 @@ private DeciderOutcome decide(final Workflow workflow, List<Task> preScheduledTa
             }
 
             Optional<TaskDef> taskDefinition = pendingTask.getTaskDefinition();
+            if (!taskDefinition.isPresent()) {
+               taskDefinition = Optional.ofNullable(workflow.getWorkflowDefinition().getTaskByRefName(pendingTask.getReferenceTaskName()))
+                       .map(WorkflowTask::getTaskDefinition);
+            }
 
             if (taskDefinition.isPresent()) {
```

Comment:
Is this if check still required?

Reply:
Yes, if no task definition is provided, it will be `Optional.empty`